src/test/ui/recursion/recursive-static-definition.stderr

```diff
@@ -4,7 +4,17 @@ error[E0391]: cycle detected when const-evaluating `FOO`
 LL | pub static FOO: u32 = FOO;
    |                       ^^^
    |
+note: ...which requires const-evaluating `FOO`...
+  --> $DIR/recursive-static-definition.rs:11:1
+   |
+LL | pub static FOO: u32 = FOO;
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: ...which again requires const-evaluating `FOO`, completing the cycle
+note: cycle used when const-evaluating + checking `FOO`
+  --> $DIR/recursive-static-definition.rs:11:1
+   |
+LL | pub static FOO: u32 = FOO;
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Similar here...

> cycle detected when const-evaluating `FOO`
> ...which requires const-evaluating `FOO`...
> ...which again requires const-evaluating `FOO`, completing the cycle

Seems strange to report a 3-element cycle when 2 would suffice?

Reply:
Oh, I guess this is because of the `Reveal` change? Those are different queries that look the same when printed?

Reply:
yes, that's the issue. We can improve the output to include the environment